Framework/src/CcdbDatabase.cxx

```diff
@@ -126,46 +126,87 @@ void CcdbDatabase::storeMO(std::shared_ptr<o2::quality_control::core::MonitorObj
   long from = getCurrentTimestamp();
   long to = getFutureTimestamp(60 * 60 * 24 * 365 * 10);
 
-  ccdbApi.storeAsTFile(mo.get(), path, metadata, from, to);
+  // extract object and metadata from MonitorObject
+  TObject* obj = mo->getObject();
+  metadata["qc_detector_name"] = mo->getDetectorName();
+  metadata["qc_task_name"] = mo->getTaskName();
+  metadata["ObjectType"] = mo->getObject()->IsA()->GetName(); // ObjectType says TObject and not MonitorObject due to a quirk in the API. Once fixed, remove this.
+
+  ccdbApi.storeAsTFileAny<TObject>(obj, path, metadata, from, to);
+}
+
+void CcdbDatabase::storeQO(std::shared_ptr<QualityObject> qo)
+{
+  // metadata
+  map<string, string> metadata;
+  // QC metadata (prefix qc_)
+  metadata["qc_version"] = Version::GetQcVersion().getString();
+  metadata["qc_quality"] = std::to_string(qo->getQuality().getLevel());
```

Comment:
Maybe also "qc_detector_name" and "qc_check_name" similarly to MOs? For example, it might be convenient for quickly filtering only the content related to one detector.

Reply:
Good idea. I did not do it because we continue storing QualityObjects but you are right that for filtering it is useful. Adding it.